identity/src/main/kotlin/net/thechance/identity/service/ResetPasswordService.kt

```diff
@@ -0,0 +1,82 @@
+package net.thechance.identity.service
+
+import net.thechance.identity.api.dto.RequestOtpResponse
+import net.thechance.identity.api.dto.VerifyOtpResponse
+import net.thechance.identity.entity.OtpLog
+import net.thechance.identity.exception.*
+import net.thechance.identity.repository.OtpLogRepository
+import net.thechance.identity.repository.UserRepository
+import net.thechance.identity.service.otpGenerator.OtpGeneratorService
+import net.thechance.identity.service.phoneNumberValidator.PhoneNumberValidatorService
+import net.thechance.identity.service.sms.SmsService
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import java.time.Instant
+import java.util.UUID
+
+@Service
+class ResetPasswordService(
+    private val phoneNumberValidatorService: PhoneNumberValidatorService,
+    private val otpGeneratorService: OtpGeneratorService,
+    private val smsService: SmsService,
+    private val otpLogRepository: OtpLogRepository,
+    private val userRepository: UserRepository
+) {
+    fun requestOtp(phoneNumber: String, defaultRegion: String): RequestOtpResponse {
+        val validatedPhoneNumber = phoneNumberValidatorService.validateAndParse(phoneNumber, defaultRegion)
+        checkPhoneNumberExistence(phoneNumber)
+        checkRequestLimitByPhoneNumber(phoneNumber)
+        expireOldActiveOtpByPhoneNumber(phoneNumber)
+        val otp = otpGeneratorService.generateOtp()
+        val otpLog = OtpLog(
+            phoneNumber = phoneNumber,
+            otp = otp,
+            expireAt = Instant.now().plusSeconds(OTP_EXPIRY_SECONDS),
+        )
```

Comment:
I suggest creating OtpService to centralize generate & verify OTP so other flows can reuse it